include/SFML/Audio/Music.hpp

```diff
@@ -50,6 +50,44 @@ class SFML_AUDIO_API Music : public SoundStream
 public:
 
     ////////////////////////////////////////////////////////////
+    /// \brief Structure defining a time range using the template type
+    ///
+    ////////////////////////////////////////////////////////////
+    template <typename T>
+    struct Span
+    {
+        ////////////////////////////////////////////////////////////
+        /// \brief Default constructor
+        ///
+        ////////////////////////////////////////////////////////////
+        Span()
+        {
+
+        }
+
+        ////////////////////////////////////////////////////////////
+        /// \brief Initialization constructor
+        ///
+        /// \param off Initial Offset
+        /// \param len Initial Length
+        ///
+        ////////////////////////////////////////////////////////////
+        Span(T off, T len):
+        offset(off),
+        length(len)
+        {
+
+        }
+
+        T offset; ///< The beginning offset of the time range
+        T length; ///< The length of the time range
+    };
+
+    // Define the relevant Span types
+    typedef Span<Time> TimeSpan;
+    typedef Span<Uint64> SampleSpan;
```

Comment:
Why is SampleSpan public if it's only used in private members? We don't even need a typename since it is used only once.